### ISSUE TITLE
Avoid Using API Lint Rule

### ISSUE THREAD

**Issue by getBoolean** [solid-software/solid_lints#81]
I've been working on a custom lint rule for banning external code by the identifier. Is this something you would want PR'd to this project?

I started working on it before custom_lint exposed the lint options, so I have some refactoring to do. I would rather add contribute it here since it is already setup for it.

**Comment by illia-romanenko** [comment]
Sure, we can take a look. Do you have any example or code available? 
Curious to know your real-world use case as well.

**Comment by getBoolean** [comment]
I've currently got it implemented in [boolean_lints](https://github.com/getBoolean/boolean_lints). I had to custom roll the configuration options when I started on it, which I plan to refactor.

For example, you can "deprecate" external code when you have a better option available:

```yaml
    banned_code:
      # Default severity
      severity: info
      entries:
        - class_name: Future
          id: wait
          source: dart:async
          reason: "Future.wait should be avoided because it looses type safety for the results. Use a Record's `wait` method instead."
          # Override severity for this entry
          severity: warning
```

Result:

```dart
void main() aync {
  await Future.wait([...]) // LINT
  await (...,).wait() // OK
}
```

**Comment by illia-romanenko** [comment]
Wow - looks really nice! If you are up for a contribution - I think we would definitely be happy having it!

**Comment by getBoolean** [comment]
Awesome, I'll start working on it and open pull request when its ready

**Comment by getBoolean** [comment]
After much hard work and some additional improvements to the lint, I have created PR #89
